peach/src/main/java/com/hartwig/hmftools/peach/LiftoverService.java

```diff
@@ -0,0 +1,205 @@
+package com.hartwig.hmftools.peach;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.hartwig.hmftools.common.genome.chromosome.Chromosome;
+import com.hartwig.hmftools.common.genome.chromosome.HumanChromosome;
+import com.hartwig.hmftools.common.genome.refgenome.RefGenomeFunctions;
+import com.hartwig.hmftools.common.utils.sv.BaseRegion;
+import htsjdk.tribble.AbstractFeatureReader;
+import htsjdk.tribble.readers.LineIterator;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFCodec;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.StringJoiner;
+import java.util.stream.Stream;
+
+import static com.hartwig.hmftools.peach.PeachUtils.BED_FILE_DELIMITER;
+import static com.hartwig.hmftools.peach.PeachUtils.PCH_LOGGER;
+import static htsjdk.tribble.AbstractFeatureReader.getFeatureReader;
+
+public class LiftoverService
+{
+    public static final String CHAIN_FILE_DELIMITER = " ";
+
+    private final PeachConfig config;
+
+    public LiftoverService(PeachConfig config)
+    {
+        this.config = config;
+    }
+
+    public void doLiftover()
+    {
+        PCH_LOGGER.info("create adjusted chain file");
+        adjustChainFile();
+
+        PCH_LOGGER.info("run Picard LiftoverVcf");
+        runPicardLiftover();
+
+        PCH_LOGGER.info("check rejected liftover variants for relevance");
+        int potentiallyMissedCount = countPotentiallyRelevantVariantsMissed();
+
+        if (potentiallyMissedCount == 0)
+        {
+            PCH_LOGGER.info("all potentially relevant variants have been lifted over");
+        }
+        else
+        {
+            PCH_LOGGER.error("some potentially relevant variants have not been lifted over: {}", potentiallyMissedCount);
+            System.exit(1);
+        }
+    }
+
+    private void adjustChainFile()
+    {
+        try (
+                Stream<String> lines = Files.lines(Paths.get(config.chainFile));
+                PrintWriter pw = new PrintWriter(config.getAdjustedChainFilePath(), StandardCharsets.UTF_8)
+        )
+        {
+            lines.forEachOrdered(line-> pw.println(getAdjustedChainFileLine(line)));
+        }
+        catch (IOException e)
+        {
+            PCH_LOGGER.error("could not create adjusted chain file: ");
+            e.printStackTrace();
```

Comment:
here you can wrap and rethrow `new RuntimeException(e)` and then exit/print stack in a central place.